server/data/incidentReportingApi.ts

```diff
@@ -229,4 +301,16 @@ export class IncidentReportingApi extends RestClient {
       path: `/incident-reports/reference/${reference}`,
     }).then(convertBasicReportDates)
   }
+
+  getReportWithDetailsById(id: string): Promise<ReportWithDetails> {
+    return this.get<DatesAsStrings<ReportWithDetails>>({
+      path: `/incident-reports/${id}/with-details`,
```

Comment:
it's probs a good idea to escape `id` for url even if we expect to have control over the contents. i mean what if a user could go to a page `/some-exciting-url/report/[id]` and that id gets passed directly to the api… a good situation would be a boring 404, a worse situation a 500.

same point with `reference` below

Reply:
https://github.com/ministryofjustice/hmpps-incident-reporting/pull/130/commits/622d47b92c3a72975a11465b59c531c864642244